### PR TITLE
Run the store gateway if using single binary

### PR DESCRIPTION
**Pull request by hi-rustin**
close https://github.com/grafana/pyroscope/issues/2045

Enable store gateway when using single binary.

### CODE REVIEW

**Comment by simonswine** [comment]
@hi-rustin thanks for your work on this, I think it is a bit more complicated depending if storage is configured or single-binary vs. micro-service mode.

https://github.com/grafana/pyroscope/issues/2045#issuecomment-1682242265

Let me know if that makes sense to you or if you think otherwise.

File: pkg/phlare/modules.go [hunk 1]
```diff
@@ -329,21 +329,15 @@ func (f *Phlare) initRing() (_ services.Service, err error) {
 
 func (f *Phlare) initStorage() (_ services.Service, err error) {
 	objectStoreTypeStats.Set(f.Cfg.Storage.Bucket.Backend)
-	if cfg := f.Cfg.Storage.Bucket; cfg.Backend != "filesystem" {
```
Comment (cyriltovena) [inline pkg/phlare/modules.go:332]: I think we should keep this for now as this won't work because of https://github.com/grafana/pyroscope/issues/2029

Comment (cyriltovena) [reply pkg/phlare/modules.go:332]: I'll check if I can resolve it.

**Comment by cyriltovena** [comment]
https://github.com/grafana/pyroscope/pull/2318

Thanks create another PR for now that will support everything but filesystem storage.

**Pull request closed without merging**